apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformer.java

```diff
@@ -0,0 +1,93 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import gov.cms.bfd.model.rda.PreAdjFissClaim;
+import gov.cms.bfd.model.rda.PreAdjFissProcCode;
+import gov.cms.bfd.pipeline.sharedutils.IdHasher;
+import gov.cms.mpsm.rda.v1.FissClaim;
+import gov.cms.mpsm.rda.v1.FissProcCodes;
+import java.time.Clock;
+
+/**
+ * Transforms a gRPC FissClaim object into a Hibernate PreAdjClaim object. Note that the gRPC data
+ * objects are not proper java beans since their optional field getters should only be called if
+ * their corresponding &quot;has&quot; methods return true. Optional fields are ignored when not
+ * present. All other fields are validated and copied into a new PreAdjFissClaim object. A
+ * lastUpdated time stamp is set using a Clock (for easier testing) and the MBI is hashed using an
+ * IdHasher.
+ */
+public class FissClaimTransformer {
+  private final Clock clock;
+  private final IdHasher idHasher;
+
+  public FissClaimTransformer(Clock clock, IdHasher idHasher) {
+    this.clock = clock;
+    this.idHasher = idHasher;
+  }
+
+  public PreAdjFissClaim transformClaim(FissClaim from) {
+    final DataTransformer transformer = new DataTransformer();
+    final PreAdjFissClaim to = new PreAdjFissClaim();
+    transformer
+        .copyString("dcn", from.getDcn(), false, 1, 23, to::setDcn)
+        .copyString("hicNo", from.getHicNo(), false, 1, 12, to::setHicNo)
+        .copyCharacter("currStatus", from.getCurrStatus(), to::setCurrStatus)
```

Comment:
We don't have an example  yet, but the "Blank" status value might come in as "" rather than " ".

Reply:
Thanks.  Good to know.  The RDA database schema had this as a non-null char.  We might need to change our schema if the value isn't actually a char.